### PR TITLE
Fixes requested by NCO on some of the resources used in operations.

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**
NCO requested clarification and updates to the resources for some of the jobs run in operations.
This PR addresses those comments and concerns.
Some of the changes are still being tested and may need further refinement.

Tagging @WeiWei-NCO

This work is part of #398 

**Type of change**

- [X] Bug fix (non-breaking change)

**How Has This Been Tested?**
The only way to test this is with NCO running the jobs in this PR and validating it meets their acceptance criteria.

  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: ecf/scripts/gdas/wave/prep/jgdas_wave_prep.ecf [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=00:10:00
-#PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128
+#PBS -l place=vscatter,select=1:mpiprocs=5:ompthreads=1:ncpus=5
 #PBS -l place=excl
```
Comment (WalterKolczynski-NOAA) [inline ecf/scripts/gdas/wave/prep/jgdas_wave_prep.ecf:8]: Does this job still need exclusive use of a node due to memory?

File: ecf/scripts/gfs/wave/prep/jgfs_wave_prep.ecf [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=00:30:00
-#PBS -l place=vscatter,select=1:ncpus=128
+#PBS -l place=vscatter,select=1:mpiprocs=60:ompthreads=1:ncpus=60
 #PBS -l place=excl
```
Comment (WalterKolczynski-NOAA) [inline ecf/scripts/gfs/wave/prep/jgfs_wave_prep.ecf]: Does this job still need to be exclusive?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.

File: ecf/scripts/gdas/wave/prep/jgdas_wave_prep.ecf (hunk 1 above)
Comment (aerorahul) [reply ecf/scripts/gdas/wave/prep/jgdas_wave_prep.ecf:8]: I don't know exactly what the memory use is in this job, hence the exclusive use.

File: ecf/scripts/gfs/wave/prep/jgfs_wave_prep.ecf (hunk 2 above)
Comment (aerorahul) [reply ecf/scripts/gfs/wave/prep/jgfs_wave_prep.ecf]: same as the gdas wave prep job.

**Comment by aerorahul** [comment]
> Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.

I haven't gone through all jobs, but we shouldn't request more CPU's than required or actually used. 
This came about from conversations w/ @WeiWei-NCO

**Comment by WalterKolczynski-NOAA** [comment]
> > Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.
> 
> I haven't gone through all jobs, but we shouldn't request more CPU's than required or actually used. This came about from conversations w/ @WeiWei-NCO

I agree. My point was just that if we are requesting exclusive mode, we are "using" the other CPUs on that node, even if we aren't requesting them explicitly.

**Comment by aerorahul** [comment]
> > > Mostly looks good as long as they meet NCO timing requirements and don't crash due to OOM. There are a couple that were reduced to less than a full node but still are requesting exclusive mode. Seems like they either shouldn't use exclusive, or you might as well throw all the CPUs at it since they are reserved anyway as long as the overhead doesn't actually make the job slower.
> > 
> > 
> > I haven't gone through all jobs, but we shouldn't request more CPU's than required or actually used. This came about from conversations w/ @WeiWei-NCO
> 
> I agree. My point was just that if we are requesting exclusive mode, we are "using" the other CPUs on that node, even if we aren't requesting them explicitly.

Actually, you are not "using" them.  If one tries to do `mpiexec -n number_exceeding_requested_cores`, the the job will fail.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf [hunk 3]
```diff
@@ -4,8 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=00:10:00
-#PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=11:ompthreads=1:ncpus=11
```
Comment (KateFriedman-NOAA) [inline ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf]: excl or memory? Wei requested every job have one of either set. The waveinit job barely needs resources so maybe `mem=1GB`?

File: ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf [hunk 4]
```diff
@@ -4,8 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=00:20:00
-#PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=8:ompthreads=1:ncpus=8
```
Comment (KateFriedman-NOAA) [inline ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf]: excl or memory? Don't know it's memory needs though.

File: ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf [hunk 5]
```diff
@@ -4,8 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=00:10:00
-#PBS -l place=vscatter,select=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=11:ompthreads=1:ncpus=11
```
Comment (KateFriedman-NOAA) [inline ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf]: Similar to my comment on the gdas_wave_init script, excl or memory?

File: ecf/scripts/gfs/wave/post/jgfs_wave_postsbs.ecf [hunk 6]
```diff
@@ -4,8 +4,7 @@
 #PBS -q %QUEUE%
 #PBS -A %PROJ%-%PROJENVIR%
 #PBS -l walltime=03:00:00
-#PBS -l place=vscatter,select=1:ncpus=128
-#PBS -l place=excl
+#PBS -l place=vscatter,select=1:mpiprocs=8:ompthreads=1:ncpus=8
```
Comment (KateFriedman-NOAA) [inline ecf/scripts/gfs/wave/post/jgfs_wave_postsbs.ecf]: excl or memory?

**Review by KateFriedman-NOAA — COMMENTED** [review]
Changes look good, I'll need to incorporate into rocoto workflow and test. Left questions on the waveinit and wavepostsbs resources changes: need excl or memory setting per Wei's request.

File: ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf (hunk 3 above)
Comment (WeiWei-NCO) [reply ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf]: 700MB on wcoss1. I would suggest to start with 2GB

File: ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf (hunk 4 above)
Comment (WeiWei-NCO) [reply ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf]: 1GB for mem

File: ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf (hunk 5 above)
Comment (WeiWei-NCO) [reply ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf]: 2GB

File: ecf/scripts/gfs/wave/post/jgfs_wave_postsbs.ecf (hunk 6 above)
Comment (WeiWei-NCO) [reply ecf/scripts/gfs/wave/post/jgfs_wave_postsbs.ecf]: 1GB

Comment (aerorahul) [reply ecf/scripts/gfs/wave/post/jgfs_wave_postsbs.ecf]: Resolved in a1ca249a

File: ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf (hunk 5 above)
Comment (aerorahul) [reply ecf/scripts/gfs/wave/init/jgfs_wave_init.ecf]: Resolved in a1ca249a

File: ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf (hunk 4 above)
Comment (aerorahul) [reply ecf/scripts/gdas/wave/post/jgdas_wave_postsbs.ecf]: Resolved in a1ca249a

File: ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf (hunk 3 above)
Comment (aerorahul) [reply ecf/scripts/gdas/wave/init/jgdas_wave_init.ecf]: Resolved in a1ca249a

**Comment by aerorahul** [comment]
@NOAA-EMC/global-workflow-cms 
I have added a few more updates since the last review following changes from NCO.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by aerorahul** [comment]
@KateFriedman-NOAA @WeiWei-NCO 
I think this is current as of this evening of testing and can be merged.
There may be a few more later, but I think we are on solid ground now.

**Review by KateFriedman-NOAA — APPROVED** [review]
Changes look good and conform to expected optimization outcomes between NCO/EMC.

**Pull request merged**